smart/model_assembly.py

```diff
@@ -293,9 +293,12 @@ def print_to_latex(
                     else:
                         df.iloc[row, col] = f"${x:0.{sig_figs}e~Lx}$"
 
-        for col in df.columns:
-            if "_" in col:
-                df = df.rename(columns={col: col.replace("_", "\_")})
+        for i, col in enumerate(df.columns):
+            if hasattr(self, "print_names"):
+                df = df.rename(columns={col: self.print_names[i]})
```

Comment:
I wonder if it would be better if `print_names` is a dictionary with `col` as key? Otherwise you limit yourself to only rename the first columns?

Reply:
I don't think that works here, `print_names` is just used to store column names that don't match the keys in the dataframe, not to store any of the other data.